apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx

```diff
@@ -81,23 +124,80 @@ export const WarehouseCollectionDetail = () => {
   const results = formatResults(queryData?.result)
 
   function loadMore() {
-    setPagination({ ...pagination, offset: pagination.offset + pagination.limit })
+    setFilters({ ...filters, offset: filters.offset + filters.limit })
   }
 
   const isLoading = queryLoading || collectionsLoading || isRefetching
 
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault()
+    setFilters({ ...filters, search })
+    refetch()
+  }
+
+  const queryUrl = `/project/${projectRef}/logs/explorer?q=${encodeURIComponent(
+    params.sql || ''
+  )}&source=warehouse`
+
   return (
     <div className="relative flex flex-col flex-grow h-full">
       <ShimmerLine active={isLoading} />
       <LoadingOpacity active={isLoading}>
         <div className="flex flex-col w-full">
-          <div className="flex justify-between items-center pr-3">
-            <h2 className="p-3">{collection?.name}</h2>
+          <div className="flex justify-between items-center h-12 px-5">
+            <form onSubmit={handleSubmit} className="flex items-center gap-2">
+              <h2 className="text-foreground-light">{collection?.name}</h2>
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <Button
+                    htmlType="submit"
+                    icon={<RefreshCcw />}
+                    type="text"
+                    loading={isLoading}
+                    disabled={isLoading}
+                    className="px-1.5"
+                  />
+                </TooltipTrigger>
+                <TooltipContent>Refresh</TooltipContent>
+              </Tooltip>
+              <Input
+                className="w-40"
```

Comment:
can probably make this longer with w-52?
![image](https://github.com/user-attachments/assets/03a4ea47-0827-4755-8adc-16893204ac56)

otherwise looks a bit squished with the placeholder
![image](https://github.com/user-attachments/assets/c0390d8b-7a7e-4018-9055-4b02f900ad39)

Reply:
we should probably add a truncate to the collection name maybe hahaha
![image](https://github.com/user-attachments/assets/db157d26-6f82-48d8-ad72-e2b5dfbaa951)

Reply:
done!